doc/protocol-notes.md

```diff
@@ -0,0 +1,42 @@
+# Self-Governed NFTs Protocol Notes
+
+This document is my notes on the paper "Self-Governed NFTs on Cardano: Meta-Standard for Standards with Vertical and Horizontal Version Travel for a Decentralized Future" by Maksymilian Brodowicz, which describes the protocol used by the Seabug NFT marketplace. I wrote this because I found the paper hard to understand; this document is meant to be referenced alongside the paper to aid understanding. Also see the code [here](https://github.com/mlabs-haskell/plutus-use-cases/tree/main/mlabs/src/Mlabs/EfficientNFT) for a reference implementation of the protocol.
+
+-   Protocol involves two separate NFTs:
+    -   "Collection NFT" aka "underlying NFT": this is created as an nft of the image to be sold
+        -   CNFT stands for collection nft
+        -   There could be multiple images in the same collection, i.e. multiple tokens under the same currency symbol
+        -   Note that there can also be different nfts for the same image
+    -   "Self-governing NFT": this is minted by "the minting policy" (section 2.1 in the paper) parameterized by the collection nft's currency symbol (as well as some other things)
+        -   sgNFT stands for self-governing nft
+        -   The sgNft's token name is a hash of the current price, owner, and the CNFT's token name
+        -   For each image nft in a collection, there can be only one sgNft corresponding to it (assuming the image token is actually an nft, i.e. it wasn't minted with multiple values and can't be minted again)
```

Comment:
given multiple CNFTs can use the same image, this isn't technically right. moreso that there can only be one sgNFT per CNFT